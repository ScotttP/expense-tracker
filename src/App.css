* {
	margin: 0;
	padding: 0;
	box-sizing: border-box;
}

html {
	scroll-behavior: smooth;
}

#content {
	padding: 0 50px 0 50px;
	height: 100vh;
	/* background-color: #3030bd; */
}
