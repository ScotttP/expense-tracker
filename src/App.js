import React, { useEffect, useState } from "react";
import {
	HashRouter as Router,
	Switch,
	Redirect,
	Route,
} from "react-router-dom";
import "./App.css";
import Account from "./Components/Account";
import Dashboard from "./Components/Dashboard";
import History from "./Components/History";
import Navbar from "./Components/Navbar/Navbar";
import Login from "./Components/Login";
import SignUp from "./Components/SignUp";
import firebase from "./FirebaseConfig";

const firebaseAuth = firebase.auth();
const firestore = firebase.firestore();

const App = () => {
	const [loggedIn, setLoggedIn] = useState(false);

	const authStateChange = (user) => {
		if (user) return setLoggedIn(true);
		else return setLoggedIn(false);
	};
	useEffect(() => {
		const unsubscribe = firebaseAuth.onAuthStateChanged(authStateChange);
		return () => {
			unsubscribe();
		};
	}, []);

	return (
		<Router>
			<Navbar currentUser={firebaseAuth.currentUser}></Navbar>
			<div id="content">
				<Switch>
					<Route
						exact
						path="/"
						render={() =>
							firebaseAuth.currentUser ? (
								<Redirect to="/Dashboard" />
							) : (
								<Login></Login>
							)
						}
					></Route>

					<Route
						exact
						path="/Login"
						render={() =>
							firebaseAuth.currentUser ? (
								<Redirect to="/Dashboard" />
							) : (
								<Login />
							)
						}
					></Route>

					<Route
						exact
						path="/SignUp"
						render={() =>
							firebaseAuth.currentUser ? (
								<Redirect to="/Dashboard" />
							) : (
								<SignUp></SignUp>
							)
						}
					></Route>
					<Route
						exact
						path="/Dashboard"
						render={() =>
							firebaseAuth.currentUser ? (
								<Dashboard></Dashboard>
							) : (
								<Redirect to="/Login" />
							)
						}
					></Route>

					<Route
						exact
						path="/History"
						render={() =>
							firebaseAuth.currentUser ? (
								<History></History>
							) : (
								<Redirect to="/Login" />
							)
						}
					></Route>
					<Route
						exact
						path="/Account"
						render={() =>
							firebaseAuth.currentUser ? (
								<Account></Account>
							) : (
								<Redirect to="/Login" />
							)
						}
					></Route>
				</Switch>
			</div>
		</Router>
	);
};

export default App;
