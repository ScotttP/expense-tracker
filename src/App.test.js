import { render, screen, cleanup, userEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { create } from "react-test-renderer";
import App from "./App";
import firebaseAuth from "./App";

// test("renders learn react link", () => {
// 	render(<App />);
// 	const linkElement = screen.getByText(/learn react/i);
// 	expect(linkElement).toBeInTheDocument();
// });

const renderWithRouter = (ui, { route = "/" } = {}) => {
	window.history.pushState({}, "Test page", route);

	return render(ui, { wrapper: MemoryRouter });
};

afterEach(cleanup);

test("currentUser is not undefined in the application", () => {
	expect(firebaseAuth.currentUser !== undefined);
});

describe("snapshot test", () => {
	test("app snapshot", () => {
		let tree = create(<App />);
		expect(tree.toJSON()).toMatchSnapshot();
	});
});

describe("route rendering test", () => {
	test("components in App render Signup Route", () => {
		renderWithRouter(<App />, { route: "/SignUp" });
		expect(screen.getByTestId("loginSignUpFormHeader")).toBeInTheDocument();
	});
	test("components in App render Login Route", () => {
		renderWithRouter(<App />, { route: "/Login" });
		expect(screen.getByTestId("loginSignUpFormHeader")).toBeInTheDocument();
	});
	test("components in App render Dashboard route", () => {
		renderWithRouter(<App />, { route: "/Dashboard" });
		expect(screen.getByTestId("loginSignUpFormHeader")).toBeInTheDocument();
		firebaseAuth.currentUser = "hi";
		expect(screen.getByText(/Hi I'm the Dashboard/i)).toBeInTheDocument();
	});
	test("components in App render History route", () => {
		renderWithRouter(<App />, { route: "/History" });
		expect(screen.getByTestId("loginSignUpFormHeader")).toBeInTheDocument();
		firebaseAuth.currentUser = "hi";
		expect(screen.getByText(/Hi I'm the History/i)).toBeInTheDocument();
	});
	test("components in App render Account route", () => {
		renderWithRouter(<App />, { route: "/Account" });
		expect(screen.getByTestId("loginSignUpFormHeader")).toBeInTheDocument();
		firebaseAuth.currentUser = "hi";
		expect(screen.getByTestId("accountRouteTest")).toBeInTheDocument();
	});
}//https://medium.com/@the_teacher/how-to-test-redirect-from-react-router-with-rtl-react-test-library-and-jest-242eced1c6b8
