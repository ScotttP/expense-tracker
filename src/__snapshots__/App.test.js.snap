// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test app snapshot 1`] = `
Array [
  <header
    className="sc-bdfBwQ fFPChT"
  >
    <div
      className="sc-gsTCUz iUqRFX"
    >
      <h1>
        Expense Tracker
      </h1>
      <span
        className="sc-hKgILt dymiNH"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-bars fa-w-14 "
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          id="fontAwesomeIcon"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
    </div>
    <ul
      className="sc-dlfnbm cLPmSc"
      display="none"
      id="Navbarlist"
    >
      <li>
        <a
          href="#/Login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
      <li>
        <a
          href="#/SignUp"
          onClick={[Function]}
        >
          SignUp
        </a>
      </li>
    </ul>
  </header>,
  <div
    id="content"
  >
    <div
      className="sc-eCssSg kHqwrC"
    >
      <form
        className="sc-jSgupP crnJYq"
      >
        <div
          className="sc-gKsewC cntkFc"
        >
          <h1
            className="sc-iBPRYJ eMDTAQ"
            data-testid="loginSignUpFormHeader"
          >
            Login
          </h1>
          <label
            className="sc-fubCfw bKdprW"
            htmlFor="loginEmailInput"
          >
            Email
            <br />
            <input
              className="sc-pFZIQ GWGWB"
              id="loginEmailInput"
              onChange={[Function]}
              type="email"
            />
          </label>
          <label
            className="sc-fubCfw bKdprW"
            htmlFor="loginPasswordInput"
          >
            <div
              className="sc-jrAGrp hNnGqS"
            >
              Password
              <a
                href="#/ForgotPassword"
                onClick={[Function]}
                style={
                  Object {
                    "color": "#3030bd",
                    "textDecoration": "none",
                  }
                }
              >
                Forgot Your Password?
              </a>
            </div>
            <input
              className="sc-pFZIQ GWGWB"
              id="loginPasswordInput"
              onChange={[Function]}
              type="password"
            />
          </label>
          <br />
          <button
            className="sc-kEjbxe gFrpFu"
            data-testid="loginSignUpButton"
            onClick={[Function]}
          >
            <b>
              Login
            </b>
          </button>
          <br />
          <p>
            or sign in with
          </p>
          <br />
          <button
            className="sc-iqHYGH bnHxCu"
            onClick={[Function]}
          >
            <img
              className="sc-crrsfI hRBQgM"
              src="icons8-google.svg"
            />
            <b>
              Google
            </b>
          </button>
          <br />
          <p
            className="sc-dQppl jzpyrc"
          >
            Don't have an account?
          </p>
          <a
            href="#/SignUp"
            onClick={[Function]}
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <b
              style={
                Object {
                  "color": "#3030bd",
                  "textDecoration": "none",
                }
              }
            >
              Sign Up
            </b>
          </a>
        </div>
      </form>
    </div>
  </div>,
]
`;
